include/small/abstract_layer.hpp

```diff
@@ -147,13 +156,153 @@ void inline compute_with_padding(dim_t H_lb, dim_t H_ub,
+
+#endif
+                    FLOAT_ABSTRACT_OP_END(op_type, op_class, step, a_cur, b_cur, c_cur, W_elements, _C_ob);
+                }
+            }
+        }
+    
+}
+
+// Edge case to handle remainder input and output channels
+// When streamlined, this is the only function that should remain
+template <typename ScalarT,
+          typename AccumT,
+          dim_t _G_b,
+          dim_t _K_b,
+          dim_t _F_cb,
+          dim_t _O_wb,
+          dim_t _stride,
+          dim_t _UNROLL, /*@todo _UNROLL should be 1, or handled if F_C_left < _UNROLL*/
+          // TODO: add a bool to switch between microkernel and default imp
+          // Leaf to describe abstract operation
+          OpType op_type,
+          int8_t op_class>
+void inline compute_with_padding(dim_t H_lb, dim_t H_ub,
+                                 dim_t W_lb, dim_t W_ub,
+                                 dim_t F_w,
+                                 dim_t W_elements,
+                                 dim_t input_col_stride,
+                                 ScalarT const *F,
+                                 ScalarT const *I,
+                                 c_tile_t *c_cur,
+                                 const dim_t F_c_left,
+                                 const dim_t G_left,
+                                 const dim_t K_left) 
+{
+#if DEBUG ==1
+    printf("_F_cb %d _G_b %d _K_b %d\n", F_c_left, _G_b, _K_b);
+
+#endif
+
+    const dim_t _C_ob = G_left * K_left;
+    const dim_t _C_ib = G_left * F_c_left;
+    const dim_t step = _stride * _C_ib;
+
+    for (uint32_t n = H_lb; n < H_ub; n++)
+    {
+        int filter_offset_h = n * F_w * F_c_left * G_left * K_left;
+        int input_stencil_h = (n - H_lb) * input_col_stride; /*+ input_col_offset + input_row_offset*/
+
+        for (uint32_t m = W_lb; m < W_ub; m++)
+        {
+            int filter_offset_w = m * F_c_left * G_left * K_left + filter_offset_h;
+            /* This is C_ib because the microkernel stretches across groups*/
+            int input_stencil_w = (m - W_lb) * _C_ib + input_stencil_h;
+
+            ScalarT const *b = F + filter_offset_w;
+            ScalarT const *a = I + input_stencil_w;
+
+            // TODO: reintroduce convolution
+            for (uint32_t ii = 0; ii < F_c_left / _UNROLL; ii++)
+            {
+
+                /// @note using platform C_ob
+                ScalarT const *b_cur = b + ii * _UNROLL * _C_ob;
+                ScalarT const *a_cur = a + ii * _UNROLL;
+
+#if DEBUG == 1 
+                printf("\t compute w padding_in channel %d\n", ii);
+                printf("input values: %f \n", a_cur[0]);
+                if (op_type == OP_CONV )
+                printf("filter values for output channel 0 : %f %f %f %f \n", b_cur[0], b_cur[1], b_cur[2], b_cur[3]);
+                printf("input col stride %d\n", input_col_stride);
+
+#endif
+                FLOAT_ABSTRACT_OP_END(op_type, op_class, step, a_cur, b_cur, c_cur, W_elements, _C_ob);
```

Comment:
I think, this is where arm/intrinsics.h is failing to compile: _C_ob is not constexpr as macro assumes (or the compiler is not advanced enough to ignore constexpr...why doesn't this fail on Zen2 platform?).